daemon/daemon.go

```diff
@@ -350,6 +350,16 @@ func (d *Daemon) Init() error {
 	return nil
 }
 
+// DegradedMode puts the daemon into an degraded mode which will
+// always return the internal error given in the "err" argument.
+//
+// This is useful to report permanent errors to the client when
+// the daemon cannot work because e.g. a selftest failed.
+func (d *Daemon) DegradedMode(err error) {
+	d.router = mux.NewRouter()
+	d.router.PathPrefix("/").Handler(InternalError(err.Error()))
```

Comment:
I wonder if we should support a least enough to run "snap version" here. Eventually we could do more things (essentially all read-only queries should work). What do you think?

Reply:
`snap debug selftest`. Jokes aside, exposing the status via `/v2/system-info` (and the actual endpoint here too) makes it useful to non-cli front ends.